### PR TITLE
Allow imperative focus on input via ref

### PR DESCRIPTION
**Pull request by asenmitrev**
- **What does this PR do?**
Enables imperative focus on the otp input for the first empty input.

- **Any background context you want to provide?**
Imperative focus is needed often when focusing inputs from outside the component. It is one of the most common uses of imperative programming in React, so it makes sense to enable it.

### CODE REVIEW

**Review by cesarvargas00 — APPROVED** [review]

**Comment by prateek3255** [comment]
Hey @asenmitrev Sorry for the delay in reviewing here, can you resolve the conflicts here I would give it a run again and then will merge if it looks good

File: src/index.tsx [hunk 1]
```diff
@@ -52,211 +52,235 @@ interface OTPInputProps {
...
+
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        focus() {
+          const inputRef = inputRefs.current[value.length] ?? inputRefs.current[0];
+          inputRef?.focus();
+        },
+      }),
+      [value, activeInput]
```
Comment (prateek3255) [inline src/index.tsx]: Why have we included both value and `activeInput` if we're only using value here in the method?

File: src/index.tsx [hunk 2]
```diff
@@ -52,211 +52,235 @@ interface OTPInputProps {
...
+      if (shouldAutoFocus) {
+        inputRefs.current[0]?.focus();
       }
+    }, [shouldAutoFocus]);
+
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        focus() {
+          const inputRef = inputRefs.current[value.length] ?? inputRefs.current[0];
```
Comment (prateek3255) [inline src/index.tsx]: Also why are we relying on the length of the value, any reason for not doing it like this:
```suggestion
          const inputRef = inputRefs.current[activeInput]
```

**Review by prateek3255 — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 1 above)
Comment (asenmitrev) [reply src/index.tsx]: Mistake, will fix

File: src/index.tsx (hunk 2 above)
Comment (asenmitrev) [reply src/index.tsx]: I tried using activeInput but the behaviour was not what seems like the best UX. Sometimes, the active input is not the first unfilled input, but a filled input before. So you would overwrite the last value of the otp when clicking focus. Using the value length allows to always focus the first unfilled input.

**Pull request closed without merging**